### PR TITLE
Versioning

### PR DESCRIPTION
**Pull request by r0fls**
add versioning to blueprints and routes

### CODE REVIEW

File: sanic/blueprints.py [hunk 1]
```diff
@@ -43,12 +44,20 @@ def register(self, app, options):
             future.handler.__blueprintname__ = self.name
             # Prepend the blueprint URI prefix if available
             uri = url_prefix + future.uri if url_prefix else future.uri
+
+            # Blueprint global version will override route version
+            if self.version is not None:
+                version = self.version
+            else:
+                version = future.version
```
Comment (yunstanford) [inline sanic/blueprints.py]:
```python
version = self.version or future.version
```

Comment (r0fls) [inline sanic/blueprints.py]: @yunstanford maybe the future version should take precedence actually? Then I can knock this comment out of the docs: https://github.com/channelcat/sanic/pull/850/files#diff-52c81980ed2f6ad30d4db74d97124d6fR32

Comment (r0fls) [inline sanic/blueprints.py]: That is the version passed from the route decorator

File: sanic/blueprints.py [hunk 2]
```diff
@@ -89,21 +95,21 @@ def register(self, app, options):
                 app.listener(event)(listener)
 
     def route(self, uri, methods=frozenset({'GET'}), host=None,
-              strict_slashes=False, stream=False):
+              strict_slashes=False, stream=False, version=None):
         """Create a blueprint route from a decorated function.
 
         :param uri: endpoint at which the route will be accessible.
         :param methods: list of acceptable HTTP methods.
```
Comment (yunstanford) [inline sanic/blueprints.py]: maybe add `:param version` in doc_string for auto-doc ?

**Comment by r0fls** [comment]
btw this was inspired by the hug API, for the record.

**Pull request merged**